### PR TITLE
IDENTITY-6536 Adding tests for identity-event

### PR DESCRIPTION
**Pull request by madurangaRA**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/EventDistributionTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package org.wso2.carbon.identity.event;
```
Comment (denuwanthi) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/EventDistributionTaskTest.java]: License Header is missing

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    }
+
+   @Test
+    public void testGetInstance() throws IdentityEventException {
+
+        IdentityEventConfigBuilder identityEventConfigBuilder =  IdentityEventConfigBuilder.getInstance();
+        assertNotNull(identityEventConfigBuilder);
+
+
```
Comment (denuwanthi) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java]: I think here we have unnecessary line spacing.

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import static org.testng.Assert.*;
+
+public class IdentityEventConfigBuilderTest {
+
+    @BeforeSuite
+    public void setup(){
+        String home = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        String config = IdentityEventConfigBuilder.class.getResource("/").getFile();
+        System.setProperty("carbon.home", home);
+        System.setProperty("carbon.config.dir.path",config);
```
Comment (nuwandiw) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java]: Better to put space after comma. When ever using comma separated parameters.

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/event/EventTest.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.testng.annotations.Test;
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventTest {
+
+    @Test
+    public void testGetEventName(){
+
+        Event event = new Event("name");
```
Comment (nuwandiw) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/event/EventTest.java]: Can't we create the `new Event("name")` instance in a setUp() method?

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/event/EventTest.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+   @Test
+    public void testEvent(){
+
+        Event event = new Event("name");
+        Assert.assertEquals(event.getEventName(),"name");
+   }
+
+   @Test
+    public void testAddEventProperty(){
```
Comment (nuwandiw) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/event/EventTest.java]: Looks like everything tested here is duplicated in `testGetEventProperties()` method.

File: components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+import org.apache.xmlbeans.impl.xb.xsdschema.Public;
+import org.junit.runner.RunWith;
+
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+import static org.testng.Assert.*;
```
Comment (omindu) [inline components/identity-event/org.wso2.carbon.identity.event/src/test/java/org.wso2.carbon.identity.event/IdentityEventConfigBuilderTest.java]: Remove .* imports

**Pull request merged**